src/wp-includes/wp-db.php

```diff
@@ -2535,6 +2537,28 @@ protected function process_fields( $table, $data, $format ) {
 		$converted_data = $this->strip_invalid_text( $data );
 
 		if ( $data !== $converted_data ) {
+
+			$problem_fields = array();
+			foreach ( $data as $field => $value ) {
+				if ( $value !== $converted_data[ $field ] ) {
+					$problem_fields[] = $field;
+				}
+			}
+
+			if ( 1 === count( $problem_fields ) ) {
+				$this->last_error = sprintf(
+					/* translators: %s Database field where the error occurred. */
+					__( 'WordPress database error: Processing the value for the following field failed: %s. The supplied value may be too long or contained invalid data.' ),
+					reset( $problem_fields )
+				);
+			} else {
+				$this->last_error = sprintf(
+					/* translators: %s Database fields where the error occurred. */
+					__( 'WordPress database error: Processing the value for the following fields failed: %s. The supplied value may be too long or contained invalid data.' ),
+					implode( ', ', $problem_fields )
+				);
+			}
```

Comment:
`_n()` should be able to handle this:

```suggestion
			$this->last_error = sprintf(
				/* translators: %s Database field(s) where the error occurred. */
				_n(
					'WordPress database error: Processing the value for the following field failed: %s. The supplied value may be too long or contained invalid data.',
					'WordPress database error: Processing the value for the following fields failed: %s. The supplied value may be too long or contained invalid data.',
					count( $problem_fields )
				),
				implode( ', ', $problem_fields )
			);
```

Reply:
See https://github.com/WordPress/wordpress-develop/pull/1874#discussion_r747962183 on why this would be incorrect

Reply:
Given the feedback to not use `_n()`, marking this `commit`.